### DISCUSSION TITLE
semantic release publish fails

### DISCUSSION THREAD

**Discussion by ShervinAbd92** [semantic-release/semantic-release#2241] [Q&A] [answered]
## Current behavior

i am trying to use automatic semantic versioning with my github action workflow. it seems that everything is working fine until the publish section. i have my workflow set at following
```yaml
jobs:
  jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.8'
    - name: install dependencies
      run: |
        python -m pip install --upgrade pip
        python -m pip install build --user
        pip install flake8 black wheel twine setuptools
        pip install .
        pip install -r requirements-dev.txt 
    - name: build a binary wheel & source tarball
      run: |
        python -m build --sdist --wheel --outdir dist/ .
    - name: twine check
      run: twine check dist/*
  release:
    runs-on: ubuntu-latest
    concurrency: release
    needs: deploy
    steps:
    - uses: actions/checkout@v2
      with:
        ref : 'main'
    - name: Python Semantic Release
      uses: relekang/python-semantic-release@master
      env:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        pypi_token: ${{ secrets.PYPI_PASSWORD }}
```

and the report i get from release part: 
```
/semantic-release/action.sh: line 3: set: -l: invalid option
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
debug: get_current_version_by_config_file()
debug: Parsing current version: path=PosixPath('setup.py') pattern='__version__ *[:=] *["\\\'](\\d+\\.\\d+(?:\\.\\d+)?)["\\\']' num_matches=1
debug: Regex matched version: 0.2.0
debug: get_current_version_by_config_file -> 0.2.0
debug: evaluate_version_bump('0.2.0', None)
debug: Reference v0.2.0 does not exist, considering entire history
debug: parse_commit_message('fix: chabged the source for semantic release')
debug: parse_commit_message -> ParsedCommit(bump=1, type='fix', scope=None, descriptions=['chabged the source for semantic release'], breaking_descriptions=[])
debug: Commits found since last release: 1
debug: evaluate_version_bump -> patch
debug: get_new_version('0.2.0', 'patch')
debug: get_new_version -> 0.2.1
debug: get_repository_owner_and_name()
debug: get_repository_owner_and_name -> ('ShervinAbd92', 'fcdproc')
debug: Running publish on branch "main"
debug: checkout('"main"')
error: Cmd('git') failed due to: exit code(1)
error:   cmdline: git checkout "main"
error:   stderr: 'error: pathspec '"main"' did not match any file(s) known to git'
```

## Expected behavior

i would have expected it publish this version bump release on github, but it fails
Any suggestion as why this is happening or how to resolve it would be appreciated?

**Comment by travi** [comment]
i updated your details by surrounding code blocks with triple backticks to make it more readable. please consider formatting when posting so that your points are easier to understand. you can find details at https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by ShervinAbd92** [reply]
sure, sorry about that.

**Comment by travi** [comment, marked as answer]
from your configuration, it looks like you are using `relekang/python-semantic-release`, which is not maintained by this team. please reach out to the maintainers of that project instead

**Reply by ShervinAbd92** [reply]
so I am not necessarily fixed on using relekang for my semantic versioning. his example came along and i used it. however, i would like to get a better understanding on how i can set up my github workflow to be able to use this semantic versioning automatically. i have a setup.py and setup.cfg file that i am using for this process. i saw example of having package.json file specifying your plugins or using npm install and then npx semantic-release. can you show me an example that works with my package structure?

**Reply by travi** [reply]
`semantic-release` can be [used with non-javascript projects](https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-use-semantic-release-to-publish-non-javascript-packages), but we do not have a python recipe available to direct you to. there are details in our [github actions recipe](https://semantic-release.gitbook.io/semantic-release/recipes/ci-configurations/github-actions) that you may find useful
